File: dehaze-python/app/router/ai_billing.py
"""AI 计费管理模块路由"""

from datetime import datetime

from fastapi import APIRouter, Body, Depends, HTTPException, Path, Query, status
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.code import ResultCode
from app.core.exceptions import BusinessException
from app.core.result import Result, success
from app.database import get_db
from app.decorators import require_permission
from app.decorators.permission import check_permission
from app.dependencies.auth import UserContext, get_current_user
from app.models.schema.ai_billing import (
    AdjustRequest,
    AnomalyRecordQuery,
    AnomalyRecordResult,
    BalanceResult,
    BillingRecordQuery,
    BillingRecordResult,
    BillingStatQuery,
    BillingStatResult,
    BillingSummaryResult,
    BillResult,
    CreditLogQuery,
    CreditLogResult,
    RefundAuditRequest,
    RefundCreateRequest,
    RefundQuery,
    RefundResult,
)
from app.models.schema.ai_billing_cost import (
    CostStatResult,
    ModelCostCreateRequest,
    ModelCostQuery,
    ModelCostResult,
    ModelCostUpdateRequest,
    ReconcileImportRequest,
)
from app.models.schema.common import PageResult
from app.service.billing.balance_service import balance_service
from app.service.billing.bill_service import bill_service
from app.service.billing.billing_anomaly_service import billing_anomaly_service
from app.service.billing.billing_record_service import billing_record_service
from app.service.billing.billing_stat_service import billing_stat_service
from app.service.billing.cost_service import cost_service
from app.service.billing.cost_stat_service import cost_stat_service
from app.service.billing.quota_service import quota_service
from app.service.billing.recharge_service import recharge_service
from app.service.billing.refund_service import refund_service

router = APIRouter(
    prefix="/api/v1/ai-billing",
    tags=["AI计费管理"],
    dependencies=[Depends(get_current_user)],
)


async def _build_balance(db: AsyncSession, user_id: int) -> BalanceResult:
    """组装余额账户视图"""
    daily_used, monthly_used = await quota_service.get_used(user_id)
    daily_limit, monthly_limit = await quota_service.get_limits(db, user_id)
    return BalanceResult(
        user_id=user_id,
        credits_balance=await balance_service.get_balance(db, user_id),
        arrears_status=await balance_service.is_arrears(user_id),
        daily_used=daily_used,
        daily_limit=daily_limit,
        monthly_used=monthly_used,
        monthly_limit=monthly_limit,
    )


def _resolve_query_user(user: UserContext, user_id_param: int | None) -> int:
    """解析查询目标用户：管理员可指定 userId 查询他人数据（需 ai:billing:stat），普通用户仅可查本人"""
    if user_id_param is None or user_id_param == user.id:
        return user.id
    if not check_permission(user, "ai:billing:stat"):
        raise HTTPException(
            status_code=status.HTTP_403_FORBIDDEN,
            detail=ResultCode.ACCESS_UNAUTHORIZED.msg,
        )
    return user_id_param


# ==================== 用户端接口 ====================


@router.get("/balance", response_model=Result[BalanceResult], summary="用户余额查询")
async def get_balance(
    userId: int | None = Query(default=None, ge=1),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(await _build_balance(db, _resolve_query_user(user, userId)))


@router.get("/summary", response_model=Result[BillingSummaryResult], summary="消耗汇总查询")
async def get_summary(
    dimension: str = Query(default="day"),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(await billing_stat_service.summary(db, user.id, dimension))


@router.get("/records", response_model=Result[PageResult[BillingRecordResult]], summary="计费明细查询")
async def list_records(
    userId: int | None = Query(default=None, ge=1),
    pageNum: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    conversationId: int | None = Query(default=None),
    billType: str | None = Query(default=None),
    modelId: str | None = Query(default=None),
    dateStart: str | None = Query(default=None),
    dateEnd: str | None = Query(default=None),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = BillingRecordQuery(
        page=pageNum,
        size=pageSize,
        conversation_id=conversationId,
        bill_type=billType,
        model_id=modelId,
        date_start=_parse_datetime(dateStart),
        date_end=_parse_datetime(dateEnd),
    )
    return success(
        await billing_record_service.list_by_user(db, _resolve_query_user(user, userId), query)
    )


@router.get("/credit-logs", response_model=Result[PageResult[CreditLogResult]], summary="余额流水查询")
async def list_credit_logs(
    userId: int | None = Query(default=None, ge=1),
    pageNum: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    source: str | None = Query(default=None),
    dateStart: str | None = Query(default=None),
    dateEnd: str | None = Query(default=None),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = CreditLogQuery(
        page=pageNum,
        size=pageSize,
        source=source,
        date_start=_parse_datetime(dateStart),
        date_end=_parse_datetime(dateEnd),
    )
    return success(
        await billing_record_service.list_credit_logs(db, _resolve_query_user(user, userId), query)
    )


@router.get("/bills/{month}", response_model=Result[BillResult], summary="月结账单查询")
async def get_bill(
    month: str = Path(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(await bill_service.get_bill(db, user.id, month))


@router.get("/bills/{month}/download", response_model=Result[BillResult], summary="账单下载")
async def download_bill(
    month: str = Path(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    # 简化实现：返回账单 JSON，前端可另存为文件（保持 JSON 信封，与账单查询一致）
    return success(await bill_service.get_bill(db, user.id, month))


@router.post("/refunds", response_model=Result[RefundResult], summary="退款申请")
async def apply_refund(
    body: RefundCreateRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(
        await refund_service.apply_refund(db, user.id, body.billing_id, body.amount, body.reason)
    )


# ==================== 管理员接口 ====================


@router.get("/refunds", response_model=Result[PageResult[RefundResult]], summary="退款申请列表")
@require_permission("ai:billing:refund")
async def list_refunds(
    userId: int | None = Query(default=None, ge=1),
    status: int | None = Query(default=None, ge=1, le=3),
    pageNum: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    dateStart: str | None = Query(default=None),
    dateEnd: str | None = Query(default=None),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = RefundQuery(
        page=pageNum,
        size=pageSize,
        status=status,
        user_id=userId,
        date_start=_parse_datetime(dateStart),
        date_end=_parse_datetime(dateEnd),
    )
    return success(await refund_service.list_refunds(db, query))


@router.get("/stats", response_model=Result[list[BillingStatResult]], summary="管理员计费统计")
@require_permission("ai:billing:stat")
async def get_stats(
    userId: int | None = Query(default=None),
    modelId: str | None = Query(default=None),
    billType: str | None = Query(default=None),
    dateStart: str | None = Query(default=None),
    dateEnd: str | None = Query(default=None),
    groupBy: str = Query(default="model"),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = BillingStatQuery(
        group_by=groupBy,
        model_id=modelId,
        bill_type=billType,
        date_start=_parse_datetime(dateStart),
        date_end=_parse_datetime(dateEnd),
    )
    # user_id 额外透传（聚合维度需要）
    return success(await billing_stat_service.stats(db, query, user_id=userId))


@router.post("/adjust", response_model=Result[BalanceResult], summary="管理员手动调整积分")
@require_permission("ai:billing:adjust")
async def adjust_credits(
    body: AdjustRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    if body.amount == 0:
        raise BusinessException(ResultCode.PARAM_ERROR, "调整积分数不能为 0")
    await recharge_service.recharge(
        db,
        body.user_id,
        body.amount,
        source="admin_adjust",
        reason=body.reason,
        operator_id=user.id,
    )
    return success(await _build_balance(db, body.user_id))


@router.post("/refunds/{refund_id}/audit", response_model=Result[RefundResult], summary="退款审核")
@require_permission("ai:billing:refund")
async def audit_refund(
    refund_id: int = Path(...),
    body: RefundAuditRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(
        await refund_service.audit_refund(
            db, refund_id, body.approved, body.audit_remark, user.id
        )
    )


@router.get("/anomalies", response_model=Result[PageResult[AnomalyRecordResult]], summary="异常计费记录查询")
@require_permission("ai:billing:stat")
async def list_anomalies(
    pageNum: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    userId: int | None = Query(default=None),
    anomalyType: str | None = Query(default=None),
    status: int | None = Query(default=None),
    dateStart: str | None = Query(default=None),
    dateEnd: str | None = Query(default=None),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = AnomalyRecordQuery(
        user_id=userId,
        anomaly_type=anomalyType,
        status=status,
        date_start=_parse_datetime(dateStart),
        date_end=_parse_datetime(dateEnd),
        page=pageNum,
        size=pageSize,
    )
    return success(await billing_anomaly_service.list_anomalies(db, query))


@router.get("/costs", response_model=Result[PageResult[ModelCostResult]], summary="成本单价列表")
@require_permission("ai:billing:cost")
async def list_costs(
    keyword: str | None = Query(default=None),
    modelId: str | None = Query(default=None),
    providerId: int | None = Query(default=None),
    pageNum: int = Query(default=1, ge=1),
    pageSize: int = Query(default=20, ge=1, le=100),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    query = ModelCostQuery(
        keyword=keyword,
        model_id=modelId,
        provider_id=providerId,
        page=pageNum,
        size=pageSize,
    )
    return success(await cost_service.list_costs(db, query))


@router.post("/costs", response_model=Result[ModelCostResult], summary="新增成本单价")
@require_permission("ai:billing:cost")
async def create_cost(
    body: ModelCostCreateRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(await cost_service.create_cost(db, body))


@router.put("/costs/{cost_id}", response_model=Result[ModelCostResult], summary="更新成本单价")
@require_permission("ai:billing:cost")
async def update_cost(
    cost_id: int = Path(...),
    body: ModelCostUpdateRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    data = body.model_dump(exclude_unset=True, exclude_none=True)
    return success(await cost_service.update_cost(db, cost_id, data))


@router.delete("/costs/{cost_id}", response_model=Result[None], summary="删除成本单价")
@require_permission("ai:billing:cost")
async def delete_cost(
    cost_id: int = Path(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    await cost_service.delete_cost(db, cost_id)
    return success(None)


@router.get("/cost-stats", response_model=Result[list[CostStatResult]], summary="成本-利润统计")
@require_permission("ai:billing:cost")
async def get_cost_stats(
    startTime: str | None = Query(default=None),
    endTime: str | None = Query(default=None),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    return success(
        await cost_stat_service.cost_stats(
            db, _parse_datetime(startTime), _parse_datetime(endTime)
        )
    )


@router.post("/reconcile/import", response_model=Result[dict], summary="供应商账单导入")
@require_permission("ai:billing:cost")
async def import_reconcile(
    body: ReconcileImportRequest = Body(...),
    db: AsyncSession = Depends(get_db),
    user: UserContext = Depends(get_current_user),
):
    imported = cost_service.import_reconcile(body.content)
    return success({"imported": imported})


def _parse_datetime(value: str | None) -> datetime | None:
    """解析前端传入的日期时间字符串，非法格式返回 None"""
    if not value:
        return None
    for fmt in ("%Y-%m-%d %H:%M:%S", "%Y-%m-%d"):
        try:
            return datetime.strptime(value, fmt)
        except ValueError:
            continue
    return None

File: dehaze-go/internal/api/ai_billing.go
package api

import (
	"strconv"

	"github.com/earthyzinc/dehaze-go/internal/model/bo"
	aiservice "github.com/earthyzinc/dehaze-go/internal/service/ai"
	"github.com/earthyzinc/dehaze-go/pkg/common"
	"github.com/earthyzinc/dehaze-go/pkg/security"
	"github.com/earthyzinc/dehaze-go/pkg/server/gin/middleware"
	"github.com/gin-gonic/gin"
)

// AiBillingApi AI 计费管理（用户端余额/明细/账单/退款申诉 + 管理端统计/调整/审核/异常/成本）。
//
// 权限校验一律放在 handler 内、参数绑定之后：FastAPI 先做请求体/查询/路径参数校验再执行
// 权限装饰器，非法参数 + 无权限的请求在 Python 端返回 A0400 而非 A0301。
type AiBillingApi struct {
	billing *aiservice.BillingService
	cost    *aiservice.CostService
}

func NewAiBillingApi(billing *aiservice.BillingService, cost *aiservice.CostService) *AiBillingApi {
	return &AiBillingApi{billing: billing, cost: cost}
}

// resolveQueryUser 解析查询目标用户：管理员可指定 userId 查询他人数据（需 ai:billing:stat），
// 普通用户仅可查本人（越权一律 A0301，与 python _resolve_query_user 一致）。
func (a *AiBillingApi) resolveQueryUser(c *gin.Context, rawUserID string) (int64, bool) {
	userID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return 0, false
	}
	if rawUserID == "" {
		return userID, true
	}
	target, parseErr := strconv.ParseInt(rawUserID, 10, 64)
	if parseErr != nil || target < 1 {
		_ = c.Error(common.NewBizError(common.PARAM_ERROR, "userId 参数非法"))
		return 0, false
	}
	if target == userID {
		return userID, true
	}
	if permErr := middleware.CheckPermission(c, "ai:billing:stat"); permErr != nil {
		_ = c.Error(permErr)
		return 0, false
	}
	return target, true
}

// ==================== 用户端 ====================

// GetBalance 用户余额查询。
func (a *AiBillingApi) GetBalance(c *gin.Context) {
	target, ok := a.resolveQueryUser(c, c.Query("userId"))
	if !ok {
		return
	}
	result, err := a.billing.Balance(c.Request.Context(), target)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// GetSummary 消耗汇总查询（dimension: day/month）。
func (a *AiBillingApi) GetSummary(c *gin.Context) {
	userID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.Summary(c.Request.Context(), userID, c.DefaultQuery("dimension", "day"))
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ListRecords 计费明细查询。
func (a *AiBillingApi) ListRecords(c *gin.Context) {
	var query bo.BillingRecordQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	// 分页口径对齐 python `Query(default=1, ge=1)` / `Query(default=20, ge=1, le=100)`：
	// 缺省取 1/20，显式传非数字、<1 或 >100 一律 A0400（不得静默回退默认值）。
	page, size, ok := parsePaginationWithSize(c, 20)
	if !ok {
		return
	}
	query.PageNum, query.PageSize = page, size
	target, ok := a.resolveQueryUser(c, c.Query("userId"))
	if !ok {
		return
	}
	result, err := a.billing.Records(c.Request.Context(), target, &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ListCreditLogs 余额流水查询。
func (a *AiBillingApi) ListCreditLogs(c *gin.Context) {
	var query bo.CreditLogQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	page, size, ok := parsePaginationWithSize(c, 20)
	if !ok {
		return
	}
	query.PageNum, query.PageSize = page, size
	target, ok := a.resolveQueryUser(c, c.Query("userId"))
	if !ok {
		return
	}
	result, err := a.billing.CreditLogs(c.Request.Context(), target, &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// GetBill 月结账单查询。
func (a *AiBillingApi) GetBill(c *gin.Context) {
	userID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.Bill(c.Request.Context(), userID, c.Param("month"))
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// DownloadBill 账单下载（保持 JSON 信封，前端另存为文件）。
func (a *AiBillingApi) DownloadBill(c *gin.Context) {
	a.GetBill(c)
}

// ApplyRefund 退款申请。
func (a *AiBillingApi) ApplyRefund(c *gin.Context) {
	userID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	var form bo.BillingRefundApplyForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.ApplyRefund(c.Request.Context(), userID, &form)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ==================== 管理端 ====================

// ListRefunds 退款申请列表。
func (a *AiBillingApi) ListRefunds(c *gin.Context) {
	var query bo.RefundQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	page, size, ok := parsePaginationWithSize(c, 20)
	if !ok {
		return
	}
	query.PageNum, query.PageSize = page, size
	if err := middleware.CheckPermission(c, "ai:billing:refund"); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.ListRefunds(c.Request.Context(), &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// AuditRefund 退款审核。
func (a *AiBillingApi) AuditRefund(c *gin.Context) {
	refundID, ok := parseID(c, "id")
	if !ok {
		return
	}
	var form bo.BillingRefundAuditForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:refund"); err != nil {
		_ = c.Error(err)
		return
	}
	operatorID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.AuditRefund(
		c.Request.Context(), refundID, *form.Approved, form.AuditRemark, operatorID)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// GetStats 管理员计费统计。
func (a *AiBillingApi) GetStats(c *gin.Context) {
	var query bo.BillingStatQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:stat"); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.Stats(c.Request.Context(), &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// AdjustCredits 管理员手动调整积分。
func (a *AiBillingApi) AdjustCredits(c *gin.Context) {
	var form bo.CreditAdjustForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:adjust"); err != nil {
		_ = c.Error(err)
		return
	}
	operatorID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.Adjust(c.Request.Context(), operatorID, &form)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ListAnomalies 异常计费记录查询。
func (a *AiBillingApi) ListAnomalies(c *gin.Context) {
	var query bo.AnomalyQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	page, size, ok := parsePaginationWithSize(c, 20)
	if !ok {
		return
	}
	query.PageNum, query.PageSize = page, size
	if err := middleware.CheckPermission(c, "ai:billing:stat"); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.billing.Anomalies(c.Request.Context(), &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ==================== 成本单价 ====================

// ListCosts 成本单价列表。
func (a *AiBillingApi) ListCosts(c *gin.Context) {
	var query bo.ModelCostQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	page, size, ok := parsePaginationWithSize(c, 20)
	if !ok {
		return
	}
	query.PageNum, query.PageSize = page, size
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.cost.ListCosts(c.Request.Context(), &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// CreateCost 新增成本单价。
func (a *AiBillingApi) CreateCost(c *gin.Context) {
	var form bo.ModelCostCreateForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	operatorID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.cost.CreateCost(c.Request.Context(), operatorID, &form)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// UpdateCost 更新成本单价。
func (a *AiBillingApi) UpdateCost(c *gin.Context) {
	costID, ok := parseID(c, "id")
	if !ok {
		return
	}
	var form bo.ModelCostUpdateForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	operatorID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.cost.UpdateCost(c.Request.Context(), costID, operatorID, &form)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// DeleteCost 删除成本单价。
func (a *AiBillingApi) DeleteCost(c *gin.Context) {
	costID, ok := parseID(c, "id")
	if !ok {
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	operatorID, err := security.RequireUserID(c)
	if err != nil {
		_ = c.Error(err)
		return
	}
	if err := a.cost.DeleteCost(c.Request.Context(), costID, operatorID); err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithMessage("一切ok", c)
}

// GetCostStats 成本-利润统计。
func (a *AiBillingApi) GetCostStats(c *gin.Context) {
	var query bo.CostStatQuery
	if err := c.ShouldBindQuery(&query); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	result, err := a.cost.CostStats(c.Request.Context(), &query)
	if err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(result, c)
}

// ImportReconcile 供应商账单导入。
func (a *AiBillingApi) ImportReconcile(c *gin.Context) {
	var form bo.ReconcileImportForm
	if err := c.ShouldBindJSON(&form); err != nil {
		_ = c.Error(err)
		return
	}
	if err := middleware.CheckPermission(c, "ai:billing:cost"); err != nil {
		_ = c.Error(err)
		return
	}
	common.OkWithData(gin.H{"imported": a.cost.ImportReconcile(&form)}, c)
}
